packages/edit-site/src/components/page-templates-template-parts/actions.js

```diff
@@ -167,8 +167,15 @@ export const deleteTemplateAction = {
 export const renameTemplateAction = {
 	id: 'rename-template',
 	label: __( 'Rename' ),
-	isEligible: ( template ) =>
-		isTemplateRemovable( template ) && template.is_custom,
+	isEligible: ( template ) => {
+		if (
+			! isTemplateRemovable( template ) ||
```

Comment:
I found this logic difficult to grok. What would you think of rewriting it by stating the conditions for each post type independently? Something along these lines could work:

```js
if (
    template.type === TEMPLATE_PART_POST_TYPE &&
    isTemplateRemovable( template )
) {
    return true;
}
if (
    template.type === TEMPLATE_POST_TYPE &&
    isTemplateRemovable( template) &&
    template.is_custom
) {
    return true;
}

return false;
```